File: problem-set-1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_pyramidHeight(void);
void printPyramid(int pyramidHeight);
void printRow(int numberOfSpaces, int numberOfBricks);

int main(void)
{
    int pyramidHeight = get_pyramidHeight();
    printPyramid(pyramidHeight);
}

int get_pyramidHeight(void)
{
    int pyramidHeight = 0;
    do
    {
        // Prompt the user for the pyramid's height
        pyramidHeight =
            get_int("Enter a positive integer value, greater than 0, for the pyramid's height: ");
    }
    while (pyramidHeight < 1);

    return pyramidHeight;
}

void printPyramid(int pyramidHeight)
{
    for (int row = 0; row < pyramidHeight; row++)
    {
        // Print row of bricks
        printRow(pyramidHeight - (row + 1), row + 1);
    }
}

void printRow(int numberOfSpaces, int numberOfBricks)
{
    // Print spaces
    for (int space = 0; space < numberOfSpaces; space++)
    {
        printf(" ");
    }

    // Print bricks
    for (int brick = 0; brick < numberOfBricks; brick++)
    {
        printf("#");
    }

    printf("\n");
}

File: problem-set-6/sentimental-mario-more/mario.py
# Prints a pyramid of a height given by user

from cs50 import get_int


def main():
    # Get height of pyramid
    height = get_height()

    # Print pyramid
    print_pyramid(height)


# Prompt user for height of pyramid
def get_height():
    while True:
        height = get_int("Enter height (Must be between 1 and 8, inclusive): ")

        # Make sure height is between 1 and 8 inclusive
        if height > 0 and height <= 8:
            return height


#  Print rows of pyramid one at a time
def print_pyramid(height):
    for current_row in range(height):
        # Print spaces for current row of left half of pyramid
        print(" " * (height - (current_row + 1)), end="")

        # Print blocks for current row of left half of pyrammid
        print("#" * (current_row + 1), end="")

        # Separate two halves of pyramid
        print("  ", end="")

        # Print blocks for current row of right half of pyramid
        print("#" * (current_row + 1))


main()
